src/session.ts

```diff
@@ -0,0 +1,160 @@
+// SPDX-License-Identifier: Apache-2.0
+"use strict";
+
+import { Signer } from "ethers";
+import { ethers } from "ethers";
+
+import { ErdstallSession, ErdstallClient } from "erdstall";
+import { Erdstall__factory } from "#erdstall/ledger/backend/contracts";
+import { TxReceipt } from "#erdstall/api/responses";
+import { BalanceProof } from "#erdstall/api/responses";
+import { ClientConfig } from "#erdstall/api/responses";
+import { Transfer, Mint, ExitRequest, TradeOffer, Trade } from "#erdstall/api/transactions";
+import { EnclaveWriter, EnclaveEvent } from "#erdstall/enclave";
+import { LedgerConnection, LedgerAdapter } from "#erdstall/ledger";
+import { ErdstallEvent } from "#erdstall/ledger";
+import { Assets } from "#erdstall/ledger/assets";
+import { Address } from "#erdstall/ledger";
+import { Uint256 } from "#erdstall/api/util";
+import { Stages } from "#erdstall/utils";
+
+export const ErrUnitialisedClient = new Error("client unitialised");
+
+export default class Session implements ErdstallSession {
```

Comment:
I think you can just `extend ErdstallClient`, then you don't need to duplicate and forward most of the calls. Also makes sense from a modelling point of view.